### ISSUE TITLE
glfs: crash when filesize is wrong

### ISSUE THREAD

**Issue by KirbyDE** [open-iscsi/tcmu-runner#361]
OS: Debian Stretch
Kernel: 4.14.13-1~bpo9+1
TCMU-Runner: 1.3.0
GlusterFS: 3.13.2-1
rtslib-fb: 2.1.fb64

>tcmu-runner[527]: tcmu_glfs_open:521 glfs/test1.img: device size and backing size disagree: device 20971518 backing 20971520
tcmu-runner[527]: 2018-01-25 00:06:42.834 527 [ERROR] tcmu_glfs_open:521 glfs/test1.img: device size and backing size disagree: device 20971518 backing 20971520
kernel: [  661.290565] show_signal_msg: 1 callbacks suppressed
kernel: [  661.290568] tcmu-runner[527]: segfault at 7f4a53dc7b88 ip 00007f4ae071b2ea sp 00007ffcb66c8e80 error 6 in libglusterfs.so.0.0.1[7f4ae06b0000+f1000]
systemd[1]: tcmu-runner.service: Main process exited, code=killed, status=11/SEGV
systemd[1]: tcmu-runner.service: Unit entered failed state.
systemd[1]: tcmu-runner.service: Failed with result 'signal'.

**Comment by mikechristie** [comment]
Adding the gluster developers.

@pkalever @pranithk

**Comment by mikechristie** [comment]
@lxbsz  I guess this is your issue now :)

**Comment by lxbsz** [comment]
Yes, sure it's mine now, and I will have a look at this later bacause I look at another gluster-block vs tcmu-runner's poor performance problem.

The bug is :https://bugzilla.redhat.com/show_bug.cgi?id=1535780, and BTW any idea about this bug ?

**Comment by lxbsz** [comment]
I will debug this from today.

**Comment by lxbsz** [comment]
This has been fixed, please see: https://github.com/open-iscsi/tcmu-runner/pull/371
